### PR TITLE
added failing test case for mid-chain cancellation

### PR DESCRIPTION
**Pull request by Artazor**
Failing test case for issue #562

### CODE REVIEW

File: test/mocha/cancel.js [hunk 1]
```diff
@@ -24,6 +24,31 @@ describe("Cancellation", function() {
         });
     });
 
+    specify("shoud be consistent for mid-chain breaking", function () {
```
Comment (bergus) [inline test/mocha/cancel.js]: I wouldn't call this "mid-chain". Yes, it cancels while the chain is currently executing, but that's basically what we have `cancel` for.
"mid-chain" means to me that you have a chain `b = a.then(…), c = b.then(…)` and `b` is cancelled.

**Comment by Artazor** [comment]
Now all is ok. Should I perform any rebase magic to the repo?

File: test/mocha/cancel.js (hunk 1 above)
Comment (Artazor) [inline test/mocha/cancel.js]: I've changed wording now it is "chain cancellation should be history-agnostic"

**Comment by Artazor** [comment]
Become irrelevant

**Pull request closed without merging**